tidy3d/plugins/design/method.py

```diff
@@ -175,28 +110,466 @@ class MethodGrid(MethodIndependent):
+        default=None,
+        title="Seed for random number generation",
+        description="Set the seed used by the optimizers to set constant random number generation.",
+    )
+
+    def any_to_int_param(self, parameter):
+        """Convert ParameterAny object to integers and provide a conversion dict to return"""
+
+        return dict(enumerate(parameter.allowed_values))
+
+    def sol_array_to_dict(
+        self, solution: np.array, keys: list, param_converter: dict
+    ) -> list[dict]:
+        """Convert an array of solutions to a list of dicts for function input"""
+        sol_dict_list = [dict(zip(keys, sol)) for sol in solution]
+
+        self._handle_param_convert(param_converter, sol_dict_list)
+
+        return sol_dict_list
+
+    def _handle_param_convert(self, param_converter, sol_dict_list):
+        for param, convert in param_converter.items():
+            for sol in sol_dict_list:
+                if isinstance(sol[param], float):
+                    sol[param] = int(round(sol[param], 0))
+                sol[param] = convert[sol[param]]
+
+
+class MethodBayOpt(MethodOptimize, ABC):
+    """A standard method for performing bayesian optimization search"""
+
+    initial_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of initial random search iterations.",
+        description="The number of search runs to be done initialially with parameter values picked randomly. This provides a starting point for the Gaussian processor to optimize from.",
+    )
+
+    n_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of bayesian optimization iterations.",
+        description="The number of iterations the Gaussian processor should be sequentially called to suggest parameter values and evaluate the results.",
+    )
+
+    acq_func: Optional[Literal["ucb", "ei", "poi"]] = pd.Field(
+        default="ucb",
+        title="Type of acquisition function.",
+        description="The style of acquisition function that should be used to suggest parameter values. More detail available `here <https://bayesian-optimization.github.io/BayesianOptimization/exploitation_vs_exploration.html>`_",
+    )
+
+    kappa: Optional[pd.PositiveFloat] = pd.Field(
+        default=2.5,
+        title="Kappa.",
+        description="The kappa coefficient used by the 'ucb' acquisition function.",
+    )
+
+    xi: Optional[pd.NonNegativeFloat] = pd.Field(
+        default=0.0,
+        title="Xi.",
+        description="The Xi coefficient used by the 'ei' and 'poi' acquisition functions",
+    )
+
+    def run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable, console) -> Tuple[Any]:
+        """Defines the search algorithm for BayOpt"""
+        try:
+            from bayes_opt import BayesianOptimization, UtilityFunction
+        except ImportError:
+            raise ImportError(
+                "Cannot run bayesian optimization as bayes_opt module not found. Please check installation."
```

Comment:
single quotes on 'bayes_opt'.  also perhaps suggest they `pip install bayes_opt` or whatever the command would be .